airbyte_cdk/sources/streams/concurrent/cursor.py

```diff
@@ -240,6 +240,15 @@ def observe(self, record: Record) -> None:
     def _extract_cursor_value(self, record: Record) -> Any:
         return self._connector_state_converter.parse_value(self._cursor_field.extract_value(record))
 
+    def close_partition_without_emit(self, partition: Partition) -> None:
```

Comment:
It feels like if we add a public method here, it should be also in the public interface. Based on that, I'm wondering if it would be easier to have the `ConcurrentCursorFactory` set the `message_repository` as a `NoopMessageRepository` to prevent the cursors from emitting anything. This solution would work with any type of cursors as well (let's say we want to support custom cursors soon)